### ISSUE TITLE
failed to use `to_msg` to convert Any to Message 

### ISSUE THREAD

**Issue by c0mm4nd** [tokio-rs/prost#921]
### Version

tonic 0.10.0 with prost 0.12.1

### Platform
latest arch linux 

### Crates

tonic 0.10.0 with prost 0.12.1

### Description

cannot convert Any to Message
```
the trait bound `my::Message: prost::name::Name` is not satisfied
the following other types implement trait `prost::name::Name`:
  prost_types::protobuf::Any
  prost_types::protobuf::Duration
  prost_types::protobuf::Timestamp
lib.rs(59, 22): required by a bound in `prost_types::<impl prost_types::protobuf::Any>::to_msg`
```

![image](https://github.com/hyperium/tonic/assets/12538215/390b88ac-0547-428c-8e77-b43923061b2b)

![image](https://github.com/hyperium/tonic/assets/12538215/80e2d30f-33be-4343-abc9-125797ae7a52)

**Comment by tarcieri** [comment]
For now you can impl the `Name` trait by hand on `my::Message`

**Comment by c0mm4nd** [comment]
Thanks for your answers.
But I have one more question, is it possible to deserialize a known struct `my::Message` from `any.value` or just a `Vec<u8>`?

**Comment by tarcieri** [comment]
@c0mm4nd you can use [`Any::to_msg`](https://docs.rs/prost-types/0.12.1/prost_types/struct.Any.html#method.to_msg) with any message type which impls `Name`, which will also validate the type URL is correct for the message type

**Comment by tarcieri** [comment]
Also this should probably be closed in favor of #926
